engine/tests/phpunit/Elgg/EntityIconServiceTest.php

```diff
@@ -822,4 +839,91 @@ public function iconDimensionsProvider() {
 		];
 	}
 
+	/**
+	 * @group IconService
+	 */
+	public function testServeIconSends400ForMalformattedRequest() {
+
+		$path_key = \Elgg\Application::GET_PATH_KEY;
+		$this->request = \Elgg\Http\Request::create("?$path_key=serve-icon/x/large");
+
+		$service = $this->createService();
+
+		$response = $service->handleServeIconRequest(false);
+
+		$this->assertEquals(400, $response->getStatusCode());
+
+		$expires = $response->getExpires();
+		$this->assertTrue(time() - $expires->modify('+1 day')->getTimestamp() < 60);
```

Comment:
Yeah, confusing. I need to write a custom assertion for this that gives wiggle room in either direction.